File: solutions/c/hamming/1/hamming.c
#include "hamming.h"
#include "stdio.h"

int compute(const char *lhs, const char *rhs)
{
    if (length(lhs) != length(rhs)) return -1;
    int diff = 0;

    for (int i = 0; lhs[i] != '\0'; i++)
    {
        if (lhs[i] != rhs[i])
        {
            diff++;
        }
    }

    return diff;
}

int length(const char *array)
{
    int length = 0;

    for (int i = 0; array[i] != '\0'; i++)
    {
        length++;
    }
    
    return length;
}

File: solutions/ruby/hamming/1/hamming.rb
=begin
Write your code for the 'Hamming' exercise in this file. Make the tests in
`hamming_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/hamming` directory.
=end

module Hamming
 def self.compute(str_A, str_B)

     if (str_A.length != str_B.length)
         raise ArgumentError.new
     end
     length = 0
     different = 0
      while length < str_A.length
          if str_A[length] != str_B[length]
              different += 1
          end
          length += 1
      end

     return different

 end


end
